src/panels/config/integrations/integration-panels/zwave_js/dialog-zwave_js-hard-reset-controller.ts

```diff
@@ -0,0 +1,156 @@
+import { HomeAssistant } from "../../../../../types";
+import { ZWaveJSHardResetControllerDialogParams } from "./show-dialog-zwave_js-hard-reset-controller";
+import { showConfirmationDialog } from "../../../../../dialogs/generic/show-dialog-box";
+import { navigate } from "../../../../../common/navigate";
+
+enum ResetStatus {
+  NotStarted,
+  InProgress,
+  Done,
+}
+
+const iconMap = {
+  [ResetStatus.NotStarted]: mdiDeleteForever,
+  [ResetStatus.InProgress]: mdiRestore,
+  [ResetStatus.Done]: mdiCheckCircle,
+};
+
+@customElement("dialog-zwave_js-hard-reset-controller")
+class DialogZWaveJSHardResetController extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _entryId?: string;
+
+  @state() private _resetStatus = ResetStatus.NotStarted;
+
+  private _subscribedDeviceAdded?: Promise<UnsubscribeFunc>;
+
+  public showDialog(params: ZWaveJSHardResetControllerDialogParams): void {
+    this._entryId = params.entryId;
+  }
+
+  public closeDialog(): void {
+    this._entryId = undefined;
+    this._resetStatus = ResetStatus.NotStarted;
+    this._unsubscribeDeviceAdded();
+
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected render() {
+    if (!this._entryId) {
+      return nothing;
+    }
+
+    return html`<ha-dialog
+      open
+      @closed=${this.closeDialog}
+      .heading=${createCloseHeading(
+        this.hass,
+        this.hass.localize(
+          `ui.panel.config.zwave_js.hard_reset_controller.${
+            ResetStatus[this._resetStatus]
+          }.title`
+        )
+      )}
+    >
+      <div class="flex-container">
+        <div>
+          <ha-svg-icon
+            .path=${iconMap[this._resetStatus]}
+            .class="icon"
+          ></ha-svg-icon>
+        </div>
+        <p>
+          ${this.hass.localize(
+            `ui.panel.config.zwave_js.hard_reset_controller.${
+              ResetStatus[this._resetStatus]
+            }.body`
+          )}
+        </p>
+      </div>
+      ${this._resetStatus === ResetStatus.NotStarted
+        ? html`<mwc-button
+              slot="primaryAction"
+              @click=${this._hardResetController}
+            >
+              ${this.hass.localize("ui.common.continue")}
+            </mwc-button>
+            <mwc-button slot="secondaryAction" @click=${this.closeDialog}>
+              ${this.hass.localize("ui.common.cancel")}
+            </mwc-button>`
+        : nothing}
+    </ha-dialog>`;
+  }
+
+  private async _hardResetController(): Promise<void> {
+    if (
+      await showConfirmationDialog(this, {
+        text: this.hass.localize(
+          `ui.panel.config.zwave_js.hard_reset_controller.confirmation`
+        ),
+        dismissText: this.hass.localize("ui.common.cancel"),
+        confirmText: this.hass.localize("ui.common.continue"),
+      })
+    ) {
+      this._subscribedDeviceAdded = subscribeHardResetControllerDeviceAdded(
```

Comment:
Why do we use a subscribe for this? Can't we just await the call?

Reply:
I originally had the call block until the reset was done but was having some issues so thought this was a better pattern. I can play around with it though later, maybe through some use of asyncio tasks I can avoid the problems on the backend

Reply:
resolved the issues so we are back to that approach